jaxopt/_src/osqp.py

```diff
@@ -599,3 +597,187 @@ def __post_init__(self):
+      Low maxiter will speedup each ADMM step but may slow down overall convergence.
+    rho_start: initial learning rate. (default: 1e-1)
+    rho_min: minimum learning rate. (default: 1e-6)
+    rho_max: maximum learning rate. (default: 1e6)
+    stepsize_updates_frequency: frequency of stepsize updates. (default: 10).
+      One every `stepsize_updates_frequency` updates computes a new stepsize.
+    primal_infeasible_tol: relative tolerance for primal infeasability detection. (default: 1e-4)
+    dual_infeasible_tol: relative tolerance for dual infeasability detection. (default: 1e-4)
+    maxiter: maximum number of iterations.  (default: 4000)
+    tol: absolute tolerance for stoping criterion (default: 1e-3).
+    termination_check_frequency: frequency of termination check. (default: 5).
+      One every `termination_check_frequency` the error is computed.
+    implicit_diff_solve: the linear system solver to use.
+  """
+
+  def __init__(self, *,
+    implicit_diff_solve: Optional[Callable] = None,
+    **kwargs):
+    if 'eq_qp_preconditioner' not in kwargs:
+      kwargs['eq_qp_preconditioner'] = 'jacobi'  # available for matrices.
+    self.box_osqp = BoxOSQP(matvec_Q=None, matvec_A=None,
+                            implicit_diff=False, **kwargs)
+    self.implicit_diff_solve = implicit_diff_solve
+
+    # same interface as CVXPY_QP => we can use the same optimality_fun.
+    self.optimality_fun = _make_cvxpy_qp_optimality_fun()
+
+    # Set up implicit diff.
+    decorator = idf.custom_root(self.optimality_fun, has_aux=True,
+                                solve=self.implicit_diff_solve)
+    # pylint: disable=g-missing-from-attributes
+    self.run = decorator(self.run)
+
+    if self.box_osqp._get_loop_options()[0]:
+      self.run = jax.jit(self.run)
+  
+  def run(self,
+          init_params: Any,
+          params_obj: Tuple[Optional[Any], Any],
+          params_eq: Optional[Tuple[Any,Any]],
+          params_ineq: Optional[Tuple[Any,Any]]) -> base.OptStep:
+    """Runs the quadratic programming solver in CVXPY.
+
+    The returned params contains both the primal and dual solutions.
+
+    Args:
+      init_params: init_params for warm_start.
+      params_obj: (Q, c).
+      params_eq: (A, b) or None if no equality constraints.
+      params_ineq: (G, h) or None if no inequality constraints.
+    Returns:
+      (params, state), ``params = (primal_var, dual_var_eq, dual_var_ineq)``
+    """
+    #TODO(lbethune): should we validate params shape like CVXPY_QP ?
+    init_params, hyper_params, eq_ineq_size = OSQP_to_BoxOSQP.fit_transform(init_params, params_obj,
+                                                                            params_eq, params_ineq)
+    sol, _ = self.box_osqp.run(init_params, **hyper_params)
+    sol = OSQP_to_BoxOSQP.inverse_transform(sol, eq_ineq_size)
+    return base.OptStep(params=sol, state=None)  # TODO(lbethune): should we return the state of box_osqp.run ?
```

Comment:
You could return an `OSQPState` object with `iter_num`, `status`, etc (removing quantities that are specific to BoxOSQP for now).